src/artifacts.js

```diff
@@ -175,8 +217,46 @@ api.declare({
   // Construct details for different storage types
   var isPublic = /^public\//.test(name);
   var details  = {};
+  let uploadId;
   switch (storageType) {
+    case 'blob':
+      // Generate the details that we'll be using.
+      details.size = input.size;
+      details.sha256 = input.sha256;
+      if (input.parts) {
```

Comment:
for size and performance, we're probably best off hashing the parts... and refer to the request body when using it... (ofcourse checking the hash)